src/zrc/api/viewsets.py

```diff
@@ -48,6 +54,29 @@ class ZaakViewSet(GeoMixin,
     filter_class = ZaakFilter
     lookup_field = 'uuid'
 
+    permission_classes = (ActionScopesRequired, ZaaktypePermission)
+    required_scopes = {
+        'list': SCOPE_ZAKEN_ALLES_LEZEN,
+        'retrieve': SCOPE_ZAKEN_ALLES_LEZEN,
+        '_zoek': SCOPE_ZAKEN_ALLES_LEZEN,
+        'create': SCOPE_ZAKEN_CREATE,
+    }
+
+    def get_queryset(self):
+        base = super().get_queryset()
+
+        # drf-yasg introspection
+        if not hasattr(self.request, 'jwt_payload'):
+            return base
+
+        if self.action == 'list':
+            zt_whitelist = self.request.jwt_payload.get('zaaktypes', [])
+            if zt_whitelist == ['*']:
+                return base  # no filtering, wildcard applies
+            return base.filter(zaaktype__in=zt_whitelist)
```

Comment:
Vraag me op dit moment af of het handig is dat claims een URL zijn. Het antwoord is waarschijnlijk ja, maar het is niet meer te begrijpen over welke zaaktypes het gaat, en ook in een auth mapper wordt dit nogal vaag.

We zouden bijna slugs moeten hebben van elk zaaktype (mor, vergunning, huwelijk, etc).

Reply:
URL is wel de enige gegarandeerde unieke referentie naar het ZT...

Reply:
I know... Of, matching gegevens (woop woop)

Reply:
Zie ook even niets anders. Dus we tackelen wel als het problematisch blijkt.